xpetra:   error: cannot define member function 'Xpetra::TpetraMap

Just got this trying to build Sierra with today's Trilinos master:

```
In file included from /scratch/mhamilt/sierra_master/objs/tpls/Trilinos/b2ed81b8c8472240/packages/xpetra/src/Xpetra_TpetraMap.hpp:2:0,
                 from /scratch/mhamilt/sierra_master/TPLs_src/Trilinos/packages/xpetra/src/Map/Xpetra_MapFactory.hpp:53,
                 from /scratch/mhamilt/sierra_master/TPLs_src/Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp:69,
                 from /scratch/mhamilt/sierra_master/TPLs_src/Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.cpp:47:
/scratch/mhamilt/sierra_master/TPLs_src/Trilinos/packages/xpetra/src/Map/Xpetra_TpetraMap_def.hpp:695:80: error: cannot define member function 'Xpetra::TpetraMap<LocalOrdinal, GlobalOrdinal, Node>::clone' within 'Xpetra::TpetraMap<int, long long int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial> >'
     TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::clone(const RCP<Node2> &node2) const
                                                                                ^~~~~
make[2]: *** [packages/xpetra/src/CMakeFiles/xpetra.dir/CrsMatrix/Xpetra_EpetraCrsMatrix.cpp.o] Error 1
make[2]: Leaving directory `/scratch/mhamilt/sierra_master/objs/tpls/Trilinos/b2ed81b8c8472240'
make[1]: *** [packages/xpetra/src/CMakeFiles/xpetra.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

```

Interestingly, I get a failure, but not the one you mention.
```
In file included from /scratch/jhu/trilinos/Trilinos/packages/xpetra/src/Vector/Xpetra_TpetraVector_def.hpp:335:
In file included from /scratch/jhu/trilinos/Trilinos/packages/xpetra/src/Headers/Xpetra_UseShortNames.hpp:49:
/scratch/jhu/trilinos/Trilinos/packages/xpetra/src/Headers/Xpetra_UseShortNamesScalar.hpp:164:73: error: member 'TpetraVector' has the same name as its class
typedef Xpetra::TpetraVector<Scalar, LocalOrdinal, GlobalOrdinal, Node> TpetraVector;
```

--- a/packages/muelu/test/maxwell/Maxwell3D.cpp
+++ b/packages/muelu/test/maxwell/Maxwell3D.cpp
@@ -48,6 +48,7 @@
 // Teuchos
 #include <Teuchos_StandardCatchMacros.hpp>
 #include <Teuchos_XMLParameterListHelpers.hpp>
+#include <Teuchos_StackedTimer.hpp>
 
 // Xpetra
 #include <Xpetra_Map.hpp>

--- a/packages/xpetra/src/Map/Xpetra_TpetraMap_def.hpp
+++ b/packages/xpetra/src/Map/Xpetra_TpetraMap_def.hpp
@@ -480,10 +480,9 @@ typename Map<LocalOrdinal, GlobalOrdinal, Node>::local_map_type TpetraMap<LocalO
     RCP<const Map<LocalOrdinal, GlobalOrdinal, Node> > replaceCommWithSubset (const Teuchos::RCP<const Teuchos::Comm<int> >& newComm) const { return Teuchos::null; }
 
 #ifdef XPETRA_ENABLE_DEPRECATED_CODE
-    template<class LocalOrdinal, class GlobalOrdinal, class Node>
     template<class Node2>
     RCP<Map<LocalOrdinal, GlobalOrdinal, Node2> > XPETRA_DEPRECATED 
-    TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::clone(const RCP<Node2> &node2) const 
+    clone(const RCP<Node2> &node2) const 
     { 
         return Teuchos::null; 
     }
@@ -689,10 +688,9 @@ typename Map<LocalOrdinal, GlobalOrdinal, Node>::local_map_type TpetraMap<LocalO
     RCP<const Map<LocalOrdinal, GlobalOrdinal, Node> > replaceCommWithSubset (const Teuchos::RCP<const Teuchos::Comm<int> >& newComm) const { return Teuchos::null; }
 
 #ifdef XPETRA_ENABLE_DEPRECATED_CODE
-    template<class LocalOrdinal, class GlobalOrdinal, class Node>
     template<class Node2>
     RCP<Map<LocalOrdinal, GlobalOrdinal, Node2> > XPETRA_DEPRECATED 
-    TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::clone(const RCP<Node2> &node2) const 
+    clone(const RCP<Node2> &node2) const 
     { 
         return Teuchos::null; 
     }

